### PR TITLE
PXB-2371 - Vulnerability: Path traversal

### PR DESCRIPTION
**Pull request by ldonoso**
https://jira.percona.com/browse/PXB-2371
https://pxb.cd.percona.com/job/percona-xtrabackup-2.4-param-medium/162

Modify xbstream so it creates/extracts streams as gnu tar does.

When creating a stream, the leading directory slashes and relative
directories are removed from the filename. Hence, when the stream is
extracted, the files will be created in the directory specified, e.i.
their paths are relative.

The parameter `-P`, or `--absolute-names`, has been added to allow
absolute paths.

It is an error if there is a file with an absolute path when extracting
a stream and the parameter `-P` has not been specified.

### CODE REVIEW

**Review by rahulmalik87 — APPROVED** [review]
LGTM

File: storage/innobase/xtrabackup/src/xbstream.c [hunk 1]
```diff
@@ -65,6 +66,7 @@ static ulong 		opt_encrypt_algo;
 static char 		*opt_encrypt_key_file = NULL;
 static char 		*opt_encrypt_key = NULL;
 static int		opt_encrypt_threads = 1;
+static int			opt_absolute_names = 0;
```
Comment (kamil-holubicki) [inline storage/innobase/xtrabackup/src/xbstream.c]: tab/space mixed?

File: storage/innobase/xtrabackup/src/xbstream.c [hunk 2]
```diff
@@ -312,6 +316,8 @@ mode_create(int argc, char **argv)
 
 	for (i = 0; i < argc; i++) {
 		char			*filepath = argv[i];
+		const char		*filepath_dst;
+		int				filepath_prefix_len;
```
Comment (kamil-holubicki) [inline storage/innobase/xtrabackup/src/xbstream.c]: formatting

File: storage/innobase/xtrabackup/src/xbstream.c [hunk 3]
```diff
@@ -100,6 +102,8 @@ static struct my_option my_long_options[] =
 	 "The default value is 1.",
 	 &opt_encrypt_threads, &opt_encrypt_threads,
 	 0, GET_INT, REQUIRED_ARG, 1, 1, INT_MAX, 0, 0, 0},
+    {"absolute-names", 'P', "Don't strip leading slashes from file names when creating archives.",
```
Comment (kamil-holubicki) [inline storage/innobase/xtrabackup/src/xbstream.c]: formatting

**Review by kamil-holubicki — APPROVED** [review]
Please fix formatting (tab/space mixed)

**Pull request merged**